Core/Core/Dashboard/K5/View/Grades/K5GradesView.swift

```diff
@@ -19,7 +19,87 @@
+        self.viewModel = viewModel
+    }
+
     var body: some View {
-        Text("Grades Details", bundle: .core)
+        VStack(spacing: 0) {
+            VStack(alignment: .leading) {
+                Button(action: {
+                    withAnimation {
+                        gradeSelectorOpen.toggle()
+                    }
+                }, label: {
+                    VStack(alignment: .leading, spacing: 0) {
+                        Text("Select", bundle: .core)
+                            .font(.regular13)
+                            .background(Color.white)
+                            .foregroundColor(.ash)
+                            .padding(.top, 15)
+                        HStack(spacing: 7) {
+                            Text(viewModel.currentGradingPeriod.title ?? "")
+                                .font(.bold24)
+                                .foregroundColor(.licorice)
+                            Image.arrowOpenDownLine
+                                .resizable()
+                                .frame(width: 12, height: 12)
+                                .foregroundColor(.licorice)
+                                .rotationEffect(.degrees(gradeSelectorOpen ? -180 : 0))
+                                .animation(.easeOut)
+                            Spacer()
+                        }
+                        .padding(.bottom, 13)
+                        Divider()
+                    }
+                })
+                .zIndex(1)
+                .background(Color.white)
+                .padding(.top, 0)
+
+                if gradeSelectorOpen {
+                    VStack(alignment: .leading) {
+                        ForEach(viewModel.gradingPeriods, id: \.self) {
+                            let gradingPeriod = $0
+                            Text(gradingPeriod.title ?? "").font(.bold20).background(Color.white).contentShape(Rectangle()).onTapGesture {
+                                viewModel.didSelect(gradingPeriod: gradingPeriod)
+                                withAnimation {
+                                    gradeSelectorOpen = false
+                                }
+                            }
+                            Divider()
+                        }
+                    }.transition(.move(edge: .top))
+                }
+            }
+            .clipped()
+            Spacer()
+            ScrollView(showsIndicators: false) {
+                CircleRefresh { endRefreshing in
+                    viewModel.refresh(completion: endRefreshing)
+                }
+                ForEach(viewModel.grades) {
+                    K5GradeCell(with: $0)
+                    Divider()
+                }
+            }
+        }
+        .padding(.horizontal)
+        .contentShape(Rectangle())
+        .onTapGesture {
+            withAnimation {
+                gradeSelectorOpen = false
+            }
+        }
+    }
+}
+
+struct K5GradesView_Previews: PreviewProvider {
```

Comment:
Let's not include preview content into release builds, please add #if DEBUG preprocessor macro.